pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeConsumptionRateManager.java

```diff
@@ -231,41 +362,87 @@ interface PartitionCountFetcher {
+   *   - Call {@link #start()} once to schedule metric emission.
+   *   - Optionally call {@link #close()} to stop the emitter.
+   * <p>
+   * Thread-safe.
    */
-  @VisibleForTesting
-  static class MetricEmitter {
+  static class AsyncMetricEmitter {
+    private static final int METRIC_EMIT_FREQUENCY_SEC = 60;
+    private final AtomicReference<Double> _rateLimit;
+    private final LongAdder _messageCount = new LongAdder();
+    private final ScheduledExecutorService _executor;
+    private final AtomicBoolean _running = new AtomicBoolean(false);
+    private final QuotaUtilizationTracker _tracker;
 
-    private final ServerMetrics _serverMetrics;
-    private final String _metricKeyName;
+    public AsyncMetricEmitter(ServerMetrics serverMetrics, String metricKeyName, double initialRateLimit) {
+      _rateLimit = new AtomicReference<>(initialRateLimit);
+      _tracker = new QuotaUtilizationTracker(serverMetrics, metricKeyName);
+      _executor = Executors.newSingleThreadScheduledExecutor(r -> {
+        Thread t = new Thread(r, "server-rate-limit-metric-emitter");
+        t.setDaemon(true);
+        return t;
+      });
+    }
 
-    // state variables
-    private long _previousMinute = -1;
-    private int _aggregateNumMessages = 0;
+    public void start() {
+      if (_running.compareAndSet(false, true)) {
+        _executor.scheduleAtFixedRate(this::emit, 0, METRIC_EMIT_FREQUENCY_SEC, TimeUnit.SECONDS);
+      }
+    }
 
-    public MetricEmitter(ServerMetrics serverMetrics, String metricKeyName) {
-      _serverMetrics = serverMetrics;
-      _metricKeyName = metricKeyName;
+    @VisibleForTesting
+    void start(int initialDelayInSeconds, int emitFrequencyInSeconds) {
+      if (_running.compareAndSet(false, true)) {
+        _executor.scheduleAtFixedRate(this::emit, initialDelayInSeconds, emitFrequencyInSeconds, TimeUnit.SECONDS);
+      }
     }
 
-    int emitMetric(int numMsgsConsumed, double rateLimit, Instant now) {
-      int ratioPercentage = 0;
-      long nowInMinutes = now.getEpochSecond() / 60;
-      if (nowInMinutes == _previousMinute) {
-        _aggregateNumMessages += numMsgsConsumed;
-      } else {
-        if (_previousMinute != -1) { // not first time
-          double actualRate = _aggregateNumMessages / ((nowInMinutes - _previousMinute) * 60.0); // messages per second
-          ratioPercentage = (int) Math.round(actualRate / rateLimit * 100);
-          _serverMetrics.setValueOfTableGauge(_metricKeyName, ServerGauge.CONSUMPTION_QUOTA_UTILIZATION,
-              ratioPercentage);
-        }
-        _aggregateNumMessages = numMsgsConsumed;
-        _previousMinute = nowInMinutes;
+    public void setRateLimit(double newRateLimit) {
+      _rateLimit.set(newRateLimit);
+    }
+
+    public void record(int numMsgsConsumed) {
+      _messageCount.add(numMsgsConsumed);
+    }
+
+    private void emit() {
+      double rateLimit = _rateLimit.get();
```

Comment:
Consider adding a try-catch and log warning if it fails to update the rate. This can help with debugging